packages/python/pyfora/typeConverters/Pandas.py

```diff
@@ -0,0 +1,169 @@
+#   Copyright 2015 Ufora Inc.
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+
+import pyfora.PureImplementationMapping as PureImplementationMapping
+
+import pandas as pd
+
+
+#######################################
+# PurePython implementations:
+#######################################
+
+class _DataframeFactory:
+    def __call__(self, data):
+        return PurePythonDataFrame(data)
+
+
+class PurePythonDataFrame:
+    """
+    TODO `columns` attr is an Index of column names in Pandas.
+    """
+    def __init__(self, data, columns=None):
+        if isinstance(data, dict):
+            self._columnNames = data.keys()
+            self._columns = [PurePythonSeries(val) for val in data.values()]
+        elif isinstance(data, list):
+            # note this for is not pandas-compliant:
+            # in pandas, a list of lists describes *rows*
+            if columns is None:
+                columns = PurePythonDataFrame._defaultNames(len(data))
+
+            self._columnNames = columns
+            self._columns = [PurePythonSeries(col) for col in data]
+        else:
+            raise Exception("haven't dealt with this case yet")
+        
+        self._numColumns = len(self._columns)
+
+        # TODO: some verification of numRows ...
+        self._numRows = len(self._columns[0])
+
+    @staticmethod
+    def _defaultNames(numNames):
+        return ["C" + str(ix) for ix in xrange(numNames)]
+
+    # in Pandas, this is a classmethod, but we don't support those yet in PurePython
+    @staticmethod
+    def from_items(items):
+        """
+        [(colname, col), ... ] -> DataFrame
+        """
+        return PurePythonDataFrame(dict(items))
```

Comment:
given the performance of our dict implementation, it might be better to construct the object directly without passing through a dict first...